### ISSUE TITLE
Can't call action with derived type

### ISSUE THREAD

**Issue by mike9182** [simple-odata-client/Simple.OData.Client#624]
I am trying to call the Action "CancelSalesOrder" which has the csdl below. 

```
<Action Name="CancelSalesOrder">
      <Parameter Name="OrderClose" Type="mscrm.crmbaseentity" Nullable="false" />
      <Parameter Name="Status" Type="Edm.Int32" Nullable="false" />
 </Action>
...
<EntityType Name="crmbaseentity" Abstract="true" />
<EntityType Name="orderclose" BaseType="mscrm.activitypointer">...</EntityType>
<EntityType Name="activitypointer" BaseType="mscrm.crmbaseentity" Abstract="true">...</EntityType>
```

**The issue seems to be that since the type of "OrderClose" is "mscrm.baseentity", which does not have any properties, Simple.OData.Client does not allow any properties to be included, even if the properties match a derived type.** I've tried several different ways of calling this action using Simple.OData.Client, however, I have not been able to find a way that works. 

**Does Simple.OData.Client support calling an action with a derived type? If so, how is this achieved?**

The action executes successfully when the following request body is sent.

```
{
    "Status": 936140011,
    "OrderClose": {
        "subject": "Close Sales Order",
        "salesorderid@odata.bind": "salesorders(5AEEC23F-DB4F-E911-A978-000D3A3611E1)",
        "@odata.type": "Microsoft.Dynamics.CRM.orderclose"
    }
}
```

I have tried the following code:
```
await client
    .Unbound()
    .Action("CancelSalesOrder")
    .Set(new Dictionary<string, object>
    {
        ["OrderClose"] = new Dictionary<string, object>
        {
            ["salesorderid"] = "salesorders(5AEEC23F-DB4F-E911-A978-000D3A3611E1)",
            ["subject"] = $"Close sales Order {DateTime.Now}",
            ["@odata.type"] = "Microsoft.Dynamics.CRM.orderclose"
        },
        ["Status"] = 936140011
    }).ExecuteAsync();
```

Which throws the following exception:
`Microsoft.OData.ODataException: 'The property 'salesorderid' does not exist on type 'Microsoft.Dynamics.CRM.crmbaseentity'. Make sure to only use property names that are defined by the type or mark the type as open type.'`

**Comment by phatcher** [comment]
According to a comment in my own code I believe this is a restriction by ODataLib - don't have an issue tag for it though

**Comment by object** [comment]
Released Released 5.7.3 with the fix from @mike9182
